earthfile2llb/converter.go

```diff
@@ -352,6 +353,60 @@ func (c *Converter) RunLocal(ctx context.Context, args []string, pushFlag bool)
 	return nil
 }
 
+// RunExitCode executes a run for the purpose of determining the exit code of the command. This can be used in conditionals.
+func (c *Converter) RunExitCode(ctx context.Context, commandName string, args, mounts, secretKeyValues []string, privileged, isWithShell, withSSH, noCache bool) (int, error) {
+	c.nonSaveCommand()
+	if !isWithShell {
+		return 0, errors.New("non-shell mode not yet supported")
+	}
+
+	// The exit code will be placed in /run. We need that dir to have been created.
+	c.mts.Final.MainState = c.mts.Final.MainState.File(
+		llb.Mkdir("/run", 0755, llb.WithParents(true)),
+		llb.WithCustomNamef("[internal] mkdir %s", "/run"))
+
+	// Perform execution, but append the command with the right shell incantation that
+	// causes it to output the exit code to a file. This is done via the shellWrap.
+	var opts []llb.RunOption
+	mountRunOpts, err := parseMounts(mounts, c.mts.Final.Target, c.mts.Final.TargetInput, c.cacheContext)
+	if err != nil {
+		return 0, errors.Wrap(err, "parse mounts")
+	}
+	opts = append(opts, mountRunOpts...)
+	if privileged {
+		opts = append(opts, llb.Security(llb.SecurityModeInsecure))
+	}
+	runStr := fmt.Sprintf(
```

Comment:
How will this work if we're in a LOCALLY target? the RunLocal doesn't use internalRun.